web/src/recent_view_ui.ts

```diff
@@ -640,13 +753,16 @@ export function update_topics_of_deleted_message_ids(message_ids) {
     }
 }
 
-export function filters_should_hide_topic(topic_data) {
+export function filters_should_hide_topic(topic_data: ConversationData): boolean {
     const msg = message_store.get(topic_data.last_msg_id);
-    const sub = sub_store.get(msg.stream_id);
+    assert(msg !== undefined);
 
-    if ((sub === undefined || !sub.subscribed) && topic_data.type === "stream") {
-        // Never try to process deactivated & unsubscribed stream msgs.
-        return true;
+    if (msg.type === "stream") {
+        const sub = sub_store.get(msg.stream_id);
+        if (!sub?.subscribed && topic_data.type === "stream") {
+            // Never try to process deactivated & unsubscribed stream msgs.
```

Comment:
I preferred spelling out `(sub === undefined || !sub.subscribed)` in our checks here, because those are two meaningfully different cases that we happen to have chosen the same behavior for. `!sub?.subscribed` just feels a bit hard to parse.

Reply:
Sounds good, changed that back

Reply:
Ah, `prettier` changed it back to `!sub?.subscribed` :/

Reply:
I guess we can live with that.